github/actions/client.go

```diff
@@ -161,48 +161,94 @@ func NewClient(ctx context.Context, githubConfigURL string, creds *ActionsAuth,
 	retryClient.HTTPClient.Transport = transport
 	ac.Client = retryClient.StandardClient()
 
-	rt, err := ac.getRunnerRegistrationToken(ctx, githubConfigURL, *creds)
+	return ac, nil
+}
+
+func (c *Client) Do(req *http.Request) (*http.Response, error) {
+	resp, err := c.Client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get runner registration token: %w", err)
+		return nil, err
 	}
 
-	adminConnInfo, err := ac.getActionsServiceAdminConnection(ctx, rt, githubConfigURL)
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = resp.Body.Close()
 	if err != nil {
-		return nil, fmt.Errorf("failed to get actions service admin connection: %w", err)
+		return nil, err
 	}
 
-	ac.ActionsServiceURL = adminConnInfo.ActionsServiceUrl
+	body = trimByteOrderMark(body)
+	resp.Body = io.NopCloser(bytes.NewReader(body))
+	return resp, nil
+}
 
-	ac.mu.Lock()
-	defer ac.mu.Unlock()
-	ac.ActionsServiceAdminToken = adminConnInfo.AdminToken
-	ac.ActionsServiceAdminTokenExpiresAt, err = actionsServiceAdminTokenExpiresAt(*adminConnInfo.AdminToken)
+func (c *Client) NewGitHubAPIRequest(ctx context.Context, method, path string, body io.Reader) (*http.Request, error) {
```

Comment:
Not at the moment, no. The idea behind having these two exported was that it can build a request if you want to do something that's not covered by the client directly (which now that I type it out I'm not sure it's a good idea anymore)